### PR TITLE
feat(core): support bundling for single non-archive file

### PR DESCRIPTION
**Pull request by hoegertn**
This PR adds a possibility to use local bundling and let the final asset be one single file that is uploaded as-is.

This can be used for several types of assets, e.g. AppSync functions.

In contrast to Lambda, these functions to not expect the asset to be a zip file.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by hoegertn** [comment]
Exemption Request:

I did not find any place where the bundling type is documented. Is this a needed addition?

File: packages/aws-cdk-lib/core/test/staging.test.ts [hunk 1]
```diff
@@ -1257,8 +1257,10 @@ describe('staging', () => {
         command: [DockerStubCommand.MULTIPLE_FILES],
         outputType: BundlingOutput.ARCHIVED,
       },
-    })).toThrow(/Bundling output directory is expected to include only a single archive file when `output` is set to `ARCHIVED`/);
+    })).toThrow(/Bundling output directory is expected to include only a single file when `output` is set to `ARCHIVED` or `SINGLE_FILE`/);
   });
+
+  // TODO test here hoegertn
```
Comment (mrgrain) [inline packages/aws-cdk-lib/core/test/staging.test.ts:1263]: 👀  ;P

File: packages/@aws-cdk-testing/framework-integ/test/aws-s3-assets/test/integ.assets.file-bundling.lit.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    asset.grantRead(user);
+  }
+}
+
+const app = new App();
+const stack = new TestStack(app, 'cdk-integ-assets-bundling');
+
+new IntegTest(app, 'IntegTest', {
+  testCases: [stack],
+});
```
Comment (mrgrain) [inline packages/@aws-cdk-testing/framework-integ/test/aws-s3-assets/test/integ.assets.file-bundling.lit.ts:37]: Any chance you can add an assertions that checks the file is in the expected location and format, as opposed to being in a zip-archive?

Comment (mrgrain) [reply packages/@aws-cdk-testing/framework-integ/test/aws-s3-assets/test/integ.assets.file-bundling.lit.ts:37]: Might be easier to test that as a unit test though.

**Review by mrgrain — DISMISSED** [review]
same comments as before

**Review by mrgrain — CHANGES_REQUESTED** [review]
@hoegertn Can you have a look at my comments and address the requested changes?

File: packages/@aws-cdk-testing/framework-integ/test/aws-s3-assets/test/integ.assets.file-bundling.lit.ts (hunk 2 above)
Comment (hoegertn) [reply packages/@aws-cdk-testing/framework-integ/test/aws-s3-assets/test/integ.assets.file-bundling.lit.ts:37]: Added a unit test

**Review by rix0rrr — APPROVED** [review]

**Review by mrgrain — APPROVED** [review]

**Comment by mrgrain** [comment]
Thanks @hoegertn ! Apologies this took a while.

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**